### DISCUSSION TITLE
State and roadmap of non-core media features

### DISCUSSION THREAD

**Discussion by joshuaboniface** [jellyfin/jellyfin-meta#41] [Technical Committees]
# Background

Discussion has been going on for quite some time (see #29) about supported media types going forward. Some of our supported types (specifically the "still" category of Books, Comics, and Photos) have serious issues that make them difficult to improve and support long-term.

# Committee Purpose

The purpose of this committee will be to decide, for each media type, what the strengths and weaknesses of the implementation are, whether there are any improvements that can be made as-is, or if not, whether to remove the media type entirely; removal does not preclude eventual addition of the type back in a post-EFCore backend.

For the purposes of this discussion, "core media features" are Movies, TV, Music, and generic Video; all other media categories defined in the codebase are eligible for discussion here.

# Committee Members

As those most involved in the discussion, I nominate @cvium, @Bond-009, and @Shadowghost with additional input requested from @thornbill and other client developers.

As per [the constitution section on technical committees](https://github.com/jellyfin/jellyfin-meta/blob/master/policies-and-procedures/jellyfin-constitution.md#technical-committees) the goal is for these 3 members to reach a consensus on each feature and its roadmap. Once a decision has been reached on each type, final approval of the action can be made and (if applicable) deprecations announced.

**Comment by oddstr13** [comment]
I am against removal of books/comics, but it really needs to be split from audio books.

**Comment by oddstr13** [comment]
I think it might be worth considering if it would make sense to allow adding media types via plugins, but this is probably not feasible until after EFcore migration is finished

**Reply by ferferga** [reply]
In that case, we should come up with a way for plugins to have "uninstall" and "install" hooks, so all the data related to that media type is removed/populated when the plugin is uninstalled/installed.

I imagine that we could extend the interface of the current JF migration system adding a "down" method (like EFCore does) to reverse a migration.

The migrations a plugin provides could also be serializable somehow, so if the user removes the DLL manually, there's no data inconsistency. Or we should somehow force/throw at boot when a plugin's DLL is missing at boot and it's "uninstall" hooks couldn't be executed.

Some other things that came to my mind:
- We must ensure OpenAPI spec can be extended by plugins. The Swagger endpoint of server must always report the up-to-date schema that server supports with it's plugins.
- Our api.jellyfin.org page must allow users to browse the "core" schema + the plugins they prefer too.
- SDKs should be extended to support this as well.

**Reply by oddstr13** [reply]
If so, we'd want the distinction between uninstall and purge, because uninstalling a plugin can be a fairly common debuging step

**Reply by iwalton3** [reply]
I agree we should not delete user data when a plugin is uninstalled by default

**Reply by crobibero** [reply]
> We must ensure OpenAPI spec can be extended by plugins. The Swagger endpoint of server must always report the up-to-date schema that server supports with it's plugins.

This is already the case

> Our api.jellyfin.org page must allow users to browse the "core" schema + the plugins they prefer too.

Doing this would be an entire project in itself as api.jellyfin.org just loads an openapi json file and is not dynamic.

I don't think being able to add media types via plugins is something we want to support- then we essentially become Kodi with infinite usecases.
